File: sbschoolmanager-api/src/main/java/com/sbschoolmanager/sbschoolmanager_api/model/Alquilar.java
package com.sbschoolmanager.sbschoolmanager_api.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import java.time.LocalDate;

@Entity
@Table(name = "ALQUILAR")
@Getter
@Setter
@NoArgsConstructor
public class Alquilar {

    @EmbeddedId
    private AlquilarId id;

    @ManyToOne
    @MapsId("codMaterial")
    @JoinColumn(name="Cod_Material_FK")
    private Material material;

    @ManyToOne
    @MapsId("codAlumno")
    @JoinColumn(name = "Cod_alumno_FK")
    private Alumno alumno;

    @Column(name = "Fecha_Inicio")
    private LocalDate fechaInicio;

    @Column(name = "Fecha_Fin")
    private LocalDate fechaFin;
}

File: sbschoolmanager_app/lib/models/alquilar.dart

import "alumno.dart";
import "material.dart";

class Alquilar{
  final Alumno alumno;
  final Material material;
  final String fechaInicio;
  final String fechaFin;

  Alquilar({
    required this.alumno,
    required this.material,
    required this.fechaInicio,
    required this.fechaFin,
  });

  factory Alquilar.fromJson(Map<String, dynamic> json){
    return Alquilar(
      alumno: Alumno.fromJson(json["alumno"]),
      material: Material.fromJson(json["material"]),
      fechaInicio: json["fechaInicio"],
      fechaFin: json["fechaFin"],
    );
  }

  Map<String , dynamic> toJson(){
    return{
      "alumno": alumno.toJson(),
      "material": material.toJson(),
      "fechaInicio": fechaInicio,
      "fechaFin": fechaFin,
    };
  }
}